dcos/mesos.py

```diff
@@ -427,6 +438,59 @@ def tasks(self, fltr=None, completed=False):
 
         return tasks
 
+    def get_container_id(self, task_id):
+        """Returns the container ID for a task ID matching `task_id`
+
+        :param task_id: The task ID which will be mapped to container ID
+        :type task_id: str
+        :returns: The container ID associated with 'task_id'
+        :rtype: str
+        """
+
+        def _get_task(task_id):
+            candidates = []
+            if 'frameworks' in self.state():
```

Comment:
How would you suggest cleaning it up? Logically I want it all inlined in this function, but then we end up going over 80 characters if I just use embedded for loops and if statements, so I broke it up into subfunctions like this.